Core/DotLanches.Application/Services/ClienteService.cs

```diff
@@ -10,12 +13,31 @@ public class ClienteService : IClienteService
             _repository = repository;
         }
 
-        public async Task Add(Cliente cliente) => await _repository.Add(cliente);
+        public async Task Add(Cliente cliente)
+        {
+            var clienteExists = await GetByCpf(cliente.Cpf!);
```

Comment:
Não poderia usar esse mesmo método da forma que está pq esse método joga uma exceção quando o cliente não existe, e no caso da criação de cliente, ele normalmente não vai encontrar.

Fiz um teste aqui e está dando erro sempre que crio um cliente novo:
![image](https://github.com/lanchesjaamp/dotlanche/assets/32016451/0d9425aa-5a95-4083-8985-e14ef2300cfa)

Reply:
Acho que o ideal seria pegar direto do repositório:
```suggestion
       var clienteExists = await _repository.GetByCpf(Validator.ValidateAndFormatCpf(cliente.Cpf!));
```